include/bls12-381/fp.hpp

```diff
@@ -19,10 +19,10 @@ class fp
     std::array<uint64_t, 6> d;
 
     fp();
-    fp(const std::array<uint64_t, 6>& d);
+    explicit fp(const std::array<uint64_t, 6>& d);
```

Comment:
Whay make it explicit?

Reply:
We only do the checks and Montgomery transform in deserialization methods. We and to reduce the probability of fp being constructed from raw array. explicit will help this slightly.

Reply:
Hum, I don't think it is very likely that you would pass a `std::array<uint64_t, 6>` without expecting to create a `fp`, but ok.